Extension/src/abTesting.ts

```diff
@@ -12,12 +12,15 @@ import * as fs from 'fs';
 const userBucketMax: number = 100;
 const userBucketString: string = "CPP.UserBucket";
 const localConfigFile: string = "cpptools.json";
+const minimumVscodeVersion: string = "1.43.2";
 
 interface Settings {
     defaultIntelliSenseEngine?: number;
     recursiveIncludes?: number;
     gotoDefIntelliSense?: number;
     enhancedColorization?: number;
+    // a record of <extensionVersion, vscodeVersion>
+    supportExtensionVscodeVersion?: Record<string, string> | null;
```

Comment:
If the property has a `?`, don't add `null` to the type declaration.

Reply:
And if we don't add an entry for every version, we shouldn't use `Record<,>` here.  `Map<,>` would be better, I think.